core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -1310,3 +1309,5 @@ private[spark] object Executor {
     }
   }
 }
+
+class KilledByTaskReaperException(message: String) extends SparkException(message)
```

Comment:
Before, it looks like we'd throw `.internalError(msg, category = "EXECUTOR")` but now we're throwing a new SparkException with no error class set, so the resulting error class now becomes `INTERNAL_ERROR` whereas before it became `INTERNAL_ERROR_EXECUTOR`

Maybe we should migrate this to the error classes framework? 

If we don't error class migrate it now, perhaps we should have this be `extends SparkException(message, cause = null, errorClass = "INTERNAL_ERROR_EXECUTOR")` so that we preserve the error class?

On the other hand, I'm not sure that this exception can even bubble to user code so in practice the exact error class might not matter: we're really just using this for control flow.

Reply:
I did not use error class since this is not user-facing, and we've been using class matching in SparkUncaughtExceptionHandler.

Reply:
Makes sense: this is effectively a control flow exception.